native/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -0,0 +1,252 @@
+                    }
+                }
+                return returnMetaData;
+            }
+        } catch (IOException e) {
+            throw new BalException("Error occurred while reading bal files!");
+        }
+        return new ArrayList<>();
+    }
+
+    private void generateClientBalFile(Entity entity) throws BalException {
+        SyntaxTree balTree = BalSyntaxTreeGenerator.generateBalFile(entity);
+        String clientPath;
+        if (entity.getModule().equals("")) {
+            clientPath = Paths.get(this.sourcePath, "modules", "generated_clients",
+                    entity.getEntityName().toLowerCase() + "_client.bal").toAbsolutePath().toString();
+        } else {
+            clientPath = Paths.get(this.sourcePath, "modules", "generated_clients",
+                            entity.getModule() + "_" + entity.getEntityName().toLowerCase() + "_client.bal")
+                    .toAbsolutePath().toString();
+        }
+        try {
+            writeOutputFile(balTree, clientPath);
+        } catch (IOException e) {
+            throw new BalException("Error occurred in formatting ballerina client object!");
+        } catch (FormatterException e) {
+            throw new BalException("Error occurred while generating ballerina client object!");
+        }
+
+    }
+    private void generateConfigurationBalFile() throws BalException {
+        SyntaxTree configTree = BalSyntaxTreeGenerator.generateConfigBalFile();
+        try {
+            writeOutputFile(configTree, Paths.get(this.sourcePath, "modules",
+                            "generated_clients", "database_configuration.bal")
+                    .toAbsolutePath().toString());
+        } catch (IOException e) {
+            throw new BalException("Error occurred in formatting database configuration file!");
+        } catch (FormatterException e) {
+            throw new BalException("Error occurred while generating database configuration file!");
+        }
+    }
+
+    private static void writeOutputFile(SyntaxTree syntaxTree, String outPath) throws IOException, FormatterException {
+        String content;
+        content = Formatter.format(syntaxTree.toSourceCode());
+        Path pathToFile = Paths.get(outPath);
+        Files.createDirectories(pathToFile.getParent());
+        try (PrintWriter writer = new PrintWriter(outPath, StandardCharsets.UTF_8.name())) {
+            writer.println(content);
+        }
+    }
+    public static boolean hasSyntacticDiagnostics(Path filePath) {
+        String content;
+        try {
+            content = Files.readString(filePath);
+        } catch (IOException e) {
+            return false;
+        }
+        TextDocument textDocument = TextDocuments.from(content);
+        return SyntaxTree.from(textDocument).hasDiagnostics();
+    }
+
+    public static boolean hasSemanticDiagnostics(Path projectPath,
+                                                 ProjectEnvironmentBuilder projectEnvironmentBuilder) {
```

Comment:
shall we move these static functions(`hasSyntacticDiagnostics`, `hasSemanticDiagnostics `) to a Utils class?

Reply:
if this functions are used inside the class, make them private

Reply:
Updated in [d6a53d3](https://github.com/ballerina-platform/persist-tools/pull/15/commits/d6a53d3f948c8f076acb3cc1f2730258a998cf40)